src/constraint_functions.jl

```diff
@@ -22,12 +22,19 @@ This function serves as fallback option if no other function is specified for a
 function constraints_capacity(m, n::Storage, 𝒯::TimeStructure, modeltype::EnergyModel)
 
     @constraint(m, [t ∈ 𝒯],
-        m[:stor_level][n, t] <= m[:stor_cap_inst][n, t]
+        m[:stor_level][n, t] <= m[:stor_level_inst][n, t]
     )
 
-    @constraint(m, [t ∈ 𝒯],
-        m[:stor_rate_use][n, t] <= m[:stor_rate_inst][n, t]
-    )
+    if has_charge(n) && isa(charge(n), UnionCapacity)
+        @constraint(m, [t ∈ 𝒯],
+            m[:stor_charge_use][n, t] <= m[:stor_charge_inst][n, t]
+        )
+    end
+    if has_discharge(n) && isa(discharge(n), UnionCapacity)
```

Comment:
might consider using dispatch for this, but ok for now I think
could consider another utility function to avoid the `isa(discharge(n), UnionCapacity)`and use something more descriptive, perhaps? Something along the lines of `has_discharge_cap(p)`

Reply:
I agree, I included it for all identification as it also simplifies the identification in potential extensions.